### ISSUE TITLE
Think about and describe hidden arrays

### ISSUE THREAD

**Issue by mik-jozef** [mik-jozef/lr-parser-typescript#8]
Think about how hidden classes whose value is an array whould work. Should they be matched with Match or MatchArr (my bet is on the former, but let's give it more thought).

- [ ] describe this in the docs
- [ ] make test cases for this

My best guess now is that a hidden class should behave exactly as if its matcher was replaced by its pattern (with 'value' replaced with the name of the matched property). A hidden class should only be matchable with Match if its value prop is matched with Match, same w/ MatchArr

**Comment by mik-jozef** [comment]
Here's a decisive argument for why hidden array classes should be matched with MatchArray, not Match. Imagine it would be the other way around:

```
new Caten(
  new MatchArr('prop', 'asdf), // This one is proper -- we're matching an array.
  new Match('prop', HiddenClass), // Inconsistency -- we're using both Match and MatchArr for the same prop.
);
```

**Comment by mik-jozef** [comment]
Fixed in v1.2.0
